tests/load/pipeline/test_query_complexity_analyzer.py

```diff
@@ -0,0 +1,180 @@
+    },
+    {
+        "query": "SELECT *, '123' AS static_val FROM my_table",
+        "complex": False,
+        "description": "star select + literal",
+    },
+    {
+        "query": "SELECT t.*, '123' AS static_val FROM my_table AS t",
+        "complex": False,
+        "description": "star select + literal + aliased table",
+    },
+    {
+        "query": "SELECT col1, col2, col3, 'static' AS static_val FROM my_table",
+        "complex": False,
+        "description": "full select + literal",
+    },
+    {
+        "query": (
+            "WITH my_table AS (SELECT * FROM my_table WHERE col1 = True) SELECT * FROM my_table;"
+        ),
+        "complex": True,
+        "description": "cte + one table ref",
+    },
+    {
+        "query": "WITH temp_table AS (SELECT * FROM my_table) SELECT * FROM temp_table;",
+        "complex": True,
+        "description": "cte + two table refs",
+    },
+    {
+        "query": "WITH x AS (SELECT a FROM y) SELECT a FROM x",
+        "complex": True,
+        "description": "cte + two table refs + sqlglot example",
+    },
+    {
+        "query": "SELECT col1, col2 FROM my_table",
+        "complex": True,
+        "description": "subset select",
+    },
+    {
+        "query": "SELECT col1, col2, '123' AS static_val FROM my_table",
+        "complex": True,
+        "description": "subset select + literal",
+    },
+    {
+        "query": "SELECT col1, col2, '123' AS col3 FROM my_table",
+        "complex": True,
+        "description": "subset select + literal as existing col",
+    },
+    {
+        "query": (
+            "SELECT * FROM my_table JOIN my_other_table ON my_table.col1 = my_other_table.col1"
+        ),
+        "complex": True,
+        "description": "join",
+    },
+    {
+        "query": "SELECT col1, col2, 'static' || col3 AS col3 FROM my_table",
+        "complex": True,
+        "description": "concat",
+    },
+    {
+        "query": "SELECT CASE WHEN col1 > 0 THEN 1 ELSE 0 END AS flag FROM my_table",
+        "complex": True,
+        "description": "case",
+    },
+    {
+        "query": "SELECT MAX(col1) AS max_val FROM my_table",
+        "complex": True,
+        "description": "aggregation",
+    },
+    {
+        "query": "SELECT col1, col2, col3, UUID() AS new_col FROM my_table",
+        "complex": True,
+        "description": "function",
+    },
+    {
+        "query": "SELECT col1 FROM my_table GROUP BY col1",
+        "complex": True,
+        "description": "group",
+    },
+    {
+        "query": "SELECT DISTINCT col1 FROM my_table",
+        "complex": True,
+        "description": "distinct",
+    },
+    {
+        "query": "SELECT ROW_NUMBER() OVER () AS row_num FROM my_table",
+        "complex": True,
+        "description": "window",
+    },
+    {
+        "query": "SELECT * FROM my_table UNION ALL SELECT * FROM my_table",
+        "complex": True,
+        "description": "union",
+    },
+]
+
+
+@pytest.mark.parametrize(
```

Comment:
I think you can remove this. It is enough to test with duckdb, alternatively you would have to supply example queries for each dialect and our assumption should be that sqlglot works the same for each language (and gives good errors if a feature is not available for one of the languages)